onnxruntime/core/graph/graph.cc

```diff
@@ -989,6 +985,20 @@ Graph::Graph(const Model& owning_model,
                      }),
       graph_mutable_nodes->end());
 
+  // For now we convert sparse_intializer to dense tensors
+  // since there are currently no supported ops that consume sparse
+  // initializers directly.
+  // Unlike Constant nodes we are not removing sparse initializers from the graph
+  // unless they are not used by any operation. This is done so we can still save space
+  // when storing an optimized model in ort format.
+  for (const auto& sparse_tensor : graph_proto_->sparse_initializer()) {
```

Comment:
Could we instead remove the sparse initializers and just keep a list of their names? When saving the model we can just convert from dense back to sparse if the name matches a sparse initializer. That way we don't waste memory on them.